### ISSUE TITLE
thread 3023610 panic: incorrect alignment

### ISSUE THREAD

**Issue by kcbark** [schmee/habu#5]
Hi!

I tried with commit e4f2d6304ccbbc7181b29dbb51fd4aa8f6a1cf85
as well as the prebuilt released binary. This is on macOS 14.2.1 (23C71)  Apple M1 but I get the following panic:

```
thread 3023610 panic: incorrect alignment
/Users/kallecarlbark/tmp/habu/src/main.zig:169:39: 0x102379aaf in materialize (habu)
        const chain_bytes: [*]Chain = @ptrCast(@alignCast(bytes[@sizeOf(ChainMeta)..]));
                                      ^
/Users/kallecarlbark/tmp/habu/src/main.zig:1080:37: 0x1023885d7 in main (habu)
            try chain_db.materialize();
                                    ^
/Users/kallecarlbark/zig/0.11.0/files/lib/std/start.zig:574:37: 0x1023991f3 in main (habu)
            const result = root.main() catch |err| {
                                    ^
???:?:?: 0x1849810df in ??? (???)
???:?:?: 0xc6127fffffffffff in ??? (???)
zsh: abort      ./zig-out/bin/habu add stuff daily
```

**Comment by schmee** [comment]
Hi @kcbark, thanks for checking out Habu and reporting this issue!

I could not reproduce this on my Intel or my M1 macbooks which are both running macOS 13, so I believe this is a macOS 14+ issue. According to the [Zig 0.11 release notes](https://ziglang.org/download/0.11.0/release-notes.html#macOS) the newest supported macOS version in that release is 13.3, so things might get funky with Zig 0.11 + macOS 14. Of course I can't confirm this on my machines, so it could also be something else.

macOS 14 support has already been added in Zig master, so I hope there will be a 0.11.1 release soon-ish which will fix the issue if it is indeed macOS 14 that is the problem. If 0.11.1 takes too long I might have to require Zig master to build Habu instead, but I'd like to avoid that for my own sanity...

All in all, sorry for the trouble! Not sure there are any workarounds at the moment, though I might set up a `dev` branch sometime soon that builds with Zig master which might fix this. If so I'll post it here 👍

**Comment by schmee** [comment]
I actually ran into this myself on one of my macOS 12 macbooks on an empty database, but I was not able to reproduce it consistently. I believe that this is fixed by https://github.com/schmee/habu/commit/4eb6357c906bc488542642c85f3bddbe307a8a74 which is included in the new [20240127 release](https://github.com/schmee/habu/releases/tag/20240127), so I'm gonna go ahead and close this issue. If you want, it would be great if you could try this release and see if it works for you, if the issue persist let me know and I'll reopen the issue straight away! 👍
